pkg/server/catalog/catalog_test.go

```diff
@@ -0,0 +1,252 @@
+package catalog
+
+import (
+	"context"
+	"errors"
+	"testing"
+	"time"
+
+	"github.com/sirupsen/logrus/hooks/test"
+	"github.com/spiffe/spire/pkg/common/catalog"
+	"github.com/spiffe/spire/pkg/common/hostservices/metricsservice"
+	"github.com/spiffe/spire/pkg/server/plugin/datastore"
+	"github.com/spiffe/spire/pkg/server/plugin/keymanager"
+	km_memory "github.com/spiffe/spire/pkg/server/plugin/keymanager/memory"
+	"github.com/spiffe/spire/pkg/server/plugin/nodeattestor"
+	na_join_token "github.com/spiffe/spire/pkg/server/plugin/nodeattestor/jointoken"
+	"github.com/spiffe/spire/pkg/server/plugin/noderesolver"
+	nr_noop "github.com/spiffe/spire/pkg/server/plugin/noderesolver/noop"
+	"github.com/spiffe/spire/pkg/server/plugin/upstreamauthority"
+	"github.com/spiffe/spire/pkg/server/plugin/upstreamca"
+	spi "github.com/spiffe/spire/proto/spire/common/plugin"
+	"github.com/spiffe/spire/test/fakes/fakeagentstore"
+	"github.com/spiffe/spire/test/fakes/fakedatastore"
+	"github.com/spiffe/spire/test/fakes/fakeidentityprovider"
+	"github.com/spiffe/spire/test/fakes/fakemetrics"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestLoad(t *testing.T) {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Minute)
+	defer cancel()
+
+	// Create a configuration with minimal required plugins
+	hclConfig := createDefaultConfig()
+
+	// Create a custom list of BuiltIn plugins, it contains the minimal necessary plugins, and fakes for Datastore, UpstreamCA and UpstreamAuthority
+	builtIn = []catalog.Plugin{
+		// Fake Datastore to prevent sql connections
+		catalog.MakePlugin("sql",
```

Comment:
I'd prefer names like `fake_ds` or something on these fakes. I don't want to give the impression we are loading real plugins.